core/src/main/scala/org/apache/spark/rdd/CoalescedRDD.scala

```diff
@@ -213,15 +210,16 @@ private class DefaultPartitionCoalescer(val balanceSlack: Double = 0.10)
   }
 
   /**
-   * Sorts and gets the least element of the list associated with key in groupHash
+   * Gets the least element of the list associated with key in groupHash
    * The returned PartitionGroup is the least loaded of all groups that represent the machine "key"
    *
    * @param key string representing a partitioned group on preferred machine key
    * @return Option of [[PartitionGroup]] that has least elements for key
    */
-  def getLeastGroupHash(key: String): Option[PartitionGroup] = {
-    groupHash.get(key).map(_.sortWith(compare).head)
-  }
+  def getLeastGroupHash(key: String): Option[PartitionGroup] =
+    groupHash
```

Comment:
What about avoiding `if` and `flatmap` and `Some()` , like:
```
    groupHash.get(key).filter(_.nonEmpty).map(_.minBy(_.numPartitions))
```

Reply:
That seems fine to me too